src/HotChocolate/Data/src/EntityFramework/Pagination/EfQueryableCursorPagingHandler.cs

```diff
@@ -103,14 +103,15 @@ private async ValueTask<Connection> SliceAsync(
                 .ToAsyncEnumerable(context.RequestAborted)
                 .ConfigureAwait(false))
             {
-                builder.Add(new Edge<TEntity>(item.Item, CursorFormatter.Format(item.Item, keys)));
-                totalCount ??= item.TotalCount;
                 fetchCount++;
 
-                if (fetchCount >= requestedCount)
+                if (fetchCount > requestedCount)
                 {
-                    break;
+                    continue;
```

Comment:
this is wrong as it would continue reading from the database. It must break.

Reply:
true, my bad should be fixed now